Add loggers to integration manifest.json

bdraco opened the pull request:
## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->

First part of https://github.com/home-assistant/architecture/discussions/715

This only implements the property and updates the manifest.json files.  The actual code to enable the debug logging will come in a future PR since this one is already large similar to #46935.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

bdraco committed:
Add loggers

bdraco committed:
Add loggers

bdraco commented:
Here is the `dict` of loggers since this is hard to review

```
{'abode': {'abodepy', 'lomond'},
 'accuweather': {'accuweather'},
 'acmeda': {'aiopulse'},
 'adax': {'adax', 'adax_local'},
 'adguard': {'adguardhome'},
 'ads': {'pyads'},
 'advantage_air': {'advantage_air'},
 'aemet': {'aemet_opendata'},
 'aftership': {'pyaftership'},
 'agent_dvr': {'agent'},
 'airly': {'airly'},
 'airnow': {'pyairnow'},
 'airthings': {'airthings'},
 'airtouch4': {'airtouch4pyapi'},
 'airvisual': {'pysmb', 'pyairvisual'},
 'aladdin_connect': {'aladdin_connect'},
 'alarmdecoder': {'adext', 'alarmdecoder'},
 'almond': {'pyalmond'},
 'alpha_vantage': {'alpha_vantage'},
 'amazon_polly': {'boto3', 's3transfer', 'botocore'},
 'ambee': {'ambee'},
 'amberelectric': {'amberelectric'},
 'ambiclimate': {'ambiclimate'},
 'ambient_station': {'aioambient'},
 'amcrest': {'amcrest'},
 'ampio': {'asmog'},
 'androidtv': {'pure_python_adb', 'androidtv', 'adb_shell'},
 'anel_pwrctrl': {'anel_pwrctrl'},
 'anthemav': {'anthemav'},
 'apache_kafka': {'kafka_python', 'aiokafka'},
 'apcupsd': {'apcaccess'},
 'apns': {'apns2', 'hyper'},
 'apple_tv': {'srptools', 'pyatv'},
 'apprise': {'apprise'},
 'aprs': {'geopy', 'geographiclib', 'aprslib'},
 'aqualogic': {'aqualogic'},
 'aquostv': {'sharp_aquos_rc'},
 'arcam_fmj': {'arcam'},
 'arlo': {'pyarlo', 'sseclient_py'},
 'arris_tg2492lg': {'arris_tg2492lg'},
 'aruba': {'pexpect', 'ptyprocess'},
 'aseko_pool_live': {'aioaseko'},
 'asterisk_mbox': {'asterisk_mbox'},
 'asuswrt': {'aioasuswrt', 'asyncssh'},
 'atag': {'pyatag'},
 'atome': {'pyatome'},
 'august': {'pubnub', 'yalexs'},
 'aurora': {'auroranoaa'},
 'aurora_abb_powerone': {'aurorapy'},
 'aussie_broadband': {'aussiebb'},
 'avea': {'avea'},
 'awair': {'python_awair'},
 'aws': {'aiobotocore', 'botocore'},
 'axis': {'axis'},
 'azure_devops': {'aioazuredevops'},
 'azure_event_hub': {'azure'},
 'azure_service_bus': {'azure'},
 'baidu': {'aip'},
 'balboa': {'pybalboa'},
 'bbb_gpio': {'Adafruit_BBIO'},
 'bbox': {'pybbox'},
 'beewi_smartclim': {'beewi_smartclim'},
 'bh1750': {'i2csense', 'smbus'},
 'bitcoin': {'blockchain'},
 'bizkaibus': {'bizkaibus'},
 'blackbird': {'pyblackbird'},
 'blebox': {'blebox_uniapi'},
 'blink': {'blinkpy'},
 'blinksticklight': {'blinkstick'},
 'blockchain': {'pyblockchain'},
 'bluetooth_le_tracker': {'pygatt'},
 'bluetooth_tracker': {'bt_proximity', 'bluetooth'},
 'bme280': {'i2csense', 'smbus', 'bme280spi'},
 'bme680': {'bme680', 'smbus'},
 'bmw_connected_drive': {'bimmer_connected'},
 'bond': {'bond_api'},
 'bosch_shc': {'boschshcpy'},
 'braviatv': {'bravia_tv'},
 'broadlink': {'broadlink'},
 'brother': {'pysmi', 'brother', 'pyasn1', 'pysnmp'},
 'brottsplatskartan': {'brottsplatskartan'},
 'brunt': {'brunt'},
 'bsblan': {'bsblan'},
 'bt_home_hub_5': {'bthomehub5_devicelist'},
 'bt_smarthub': {'btsmarthub_devicelist'},
 'buienradar': {'buienradar', 'vincenty'},
 'caldav': {'caldav', 'vobject'},
 'canary': {'canary'},
 'cast': {'pychromecast', 'casttube'},
 'channels': {'pychannels'},
 'circuit': {'circuit_webhook'},
 'cisco_ios': {'pexpect', 'ptyprocess'},
 'cisco_mobility_express': {'ciscomobilityexpress'},
 'cisco_webex_teams': {'webexteamssdk'},
 'clementine': {'clementineremote'},
 'climacell': {'pyclimacell'},
 'cloud': {'hass_nabucasa'},
 'cloudflare': {'pycfdns'},
 'cmus': {'pycmus', 'pbr'},
 'co2signal': {'CO2Signal'},
 'coinbase': {'coinbase'},
 'comfoconnect': {'pycomfoconnect'},
 'concord232': {'stevedore', 'concord232'},
 'control4': {'pyControl4'},
 'coolmaster': {'pycoolmasternet_async'},
 'coronavirus': {'coronavirus'},
 'cpuspeed': {'cpuinfo'},
 'crownstone': {'crownstone_cloud',
                'crownstone_core',
                'crownstone_sse',
                'crownstone_uart'},
 'daikin': {'pydaikin'},
 'danfoss_air': {'pydanfossair'},
 'darksky': {'forecastio'},
 'datadog': {'datadog'},
 'deconz': {'pydeconz'},
 'decora': {'decora', 'bluepy'},
 'decora_wifi': {'decora_wifi'},
 'delijn': {'pydelijn'},
 'deluge': {'deluge_client'},
 'denonavr': {'denonavr'},
 'deutsche_bahn': {'schiene'},
 'devolo_home_control': {'devolo_home_control_api'},
 'devolo_home_network': {'devolo_plc_api'},
 'dexcom': {'pydexcom'},
 'dhcp': {'aiodiscover', 'pyroute2', 'scapy', 'dnspython'},
 'digital_ocean': {'digitalocean'},
 'digitalloggers': {'dlipower'},
 'directv': {'directv'},
 'discogs': {'discogs_client'},
 'discord': {'discord'},
 'discovery': {'netdisco'},
 'dlib_face_detect': {'face_recognition'},
 'dlib_face_identify': {'face_recognition'},
 'dlink': {'pyW215'},
 'dlna_dmr': {'async_upnp_client'},
 'dominos': {'pizzapi'},
 'doods': {'pydoods'},
 'doorbird': {'doorbirdpy'},
 'dsmr': {'dsmr_parser'},
 'dunehd': {'pdunehd'},
 'dwd_weather_warnings': {'dwdwfsapi'},
 'dweet': {'dweepy'},
 'dynalite': {'dynalite_devices_lib'},
 'eafm': {'aioeafm'},
 'ebox': {'pyebox'},
 'ebusd': {'ebusdpy'},
 'ecoal_boiler': {'ecoaliface'},
 'ecobee': {'pyecobee'},
 'econet': {'paho_mqtt', 'pyeconet'},
 'ecovacs': {'sucks', 'sleekxmppfs'},
 'eddystone_temperature': {'beacontools'},
 'edimax': {'pyedimax'},
 'edl21': {'sml'},
 'efergy': {'pyefergy', 'iso4217'},
 'egardia': {'pythonegardia'},
 'eight_sleep': {'pyeight'},
 'elgato': {'elgato'},
 'elkm1': {'elkm1_lib'},
 'elmax': {'elmax_api'},
 'elv': {'pypca'},
 'emby': {'pyemby'},
 'emonitor': {'aioemonitor'},
 'emulated_kasa': {'sense_energy'},
 'emulated_roku': {'emulated_roku'},
 'enigma2': {'openwebif'},
 'enocean': {'enocean'},
 'enphase_envoy': {'envoy_reader'},
 'entur_public_transport': {'enturclient'},
 'environment_canada': {'env_canada'},
 'envisalink': {'pyenvisalink'},
 'ephember': {'pyephember'},
 'epson': {'epson_projector'},
 'epsonworkforce': {'epsonprinter_pkg'},
 'eq3btsmart': {'eq3bt', 'bluepy'},
 'esphome': {'aioesphomeapi', 'noiseprotocol'},
 'etherscan': {'pyetherscan'},
 'eufy': {'lakeside'},
 'everlights': {'pyeverlights'},
 'evil_genius_labs': {'pyevilgenius'},
 'evohome': {'evohomeasync2', 'evohomeasync'},
 'ezviz': {'paho_mqtt', 'pyezviz'},
 'faa_delays': {'faadelays'},
 'familyhub': {'pyfamilyhublocal'},
 'fastdotcom': {'fastdotcom'},
 'feedreader': {'feedparser', 'sgmllib3k'},
 'fibaro': {'fiblary3'},
 'fido': {'pyfido'},
 'fints': {'sepaxml', 'fints', 'mt_940'},
 'fireservicerota': {'pyfireservicerota'},
 'firmata': {'pymata_express'},
 'fitbit': {'fitbit'},
 'fixer': {'fixerio'},
 'fjaraskupan': {'fjaraskupan', 'bleak'},
 'fleetgo': {'ritassist', 'geopy'},
 'flic': {'pyflic'},
 'flick_electric': {'pyflick'},
 'flipr': {'flipr_api'},
 'flo': {'aioflo'},
 'flume': {'pyflume'},
 'flunearyou': {'pyflunearyou'},
 'flux_led': {'flux_led'},
 'folder_watcher': {'watchdog'},
 'foobot': {'foobot_async'},
 'forecast_solar': {'forecast_solar'},
 'forked_daapd': {'pyforked_daapd', 'pylibrespot_java'},
 'fortios': {'paramiko', 'fortiosapi'},
 'foscam': {'libpyfoscam'},
 'free_mobile': {'freesms'},
 'freebox': {'freebox_api'},
 'freedompro': {'pyfreedompro'},
 'fritz': {'fritzconnection'},
 'fritzbox': {'pyfritzhome'},
 'fritzbox_callmonitor': {'fritzconnection'},
 'fronius': {'pyfronius'},
 'frontier_silicon': {'afsapi'},
 'futurenow': {'pyfnip'},
 'garages_amsterdam': {'garages_amsterdam'},
 'gdacs': {'aio_georss_gdacs'},
 'geniushub': {'geniushubclient'},
 'geo_json_events': {'geojson_client'},
 'geo_rss_events': {'georss_generic_client', 'georss_client'},
 'geonetnz_quakes': {'aio_geojson_geonetnz_quakes'},
 'geonetnz_volcano': {'aio_geojson_geonetnz_volcano'},
 'gios': {'dacite', 'gios'},
 'github': {'aiogithubapi'},
 'gitlab_ci': {'gitlab'},
 'gitter': {'gitterpy'},
 'glances': {'glances_api'},
 'gntp': {'gntp'},
 'goalfeed': {'pysher'},
 'goalzero': {'goalzero'},
 'gogogate2': {'ismartgate'},
 'goodwe': {'goodwe'},
 'google': {'googleapiclient'},
 'google_maps': {'locationsharinglib'},
 'google_translate': {'gtts'},
 'google_travel_time': {'googlemaps'},
 'gpsd': {'gps3'},
 'gree': {'greeclimate'},
 'greeneye_monitor': {'greeneye'},
 'greenwave': {'greenwavereality'},
 'growatt_server': {'growattServer'},
 'gstreamer': {'gsp'},
 'gtfs': {'pygtfs'},
 'guardian': {'aioguardian'},
 'habitica': {'habitipy', 'plumbum'},
 'hangouts': {'hangups', 'urwid'},
 'harman_kardon_avr': {'hkavr'},
 'harmony': {'aioharmony', 'slixmpp'},
 'hdmi_cec': {'pycec'},
 'heatmiser': {'heatmiserV3'},
 'heos': {'pyheos'},
 'here_travel_time': {'herepy'},
 'hikvision': {'pyhik'},
 'hikvisioncam': {'hikvision'},
 'hisense_aehw4a1': {'pyaehw4a1'},
 'hive': {'apyhiveapi'},
 'hlk_sw16': {'hlk_sw16'},
 'home_connect': {'homeconnect'},
 'home_plus_control': {'homepluscontrol'},
 'homekit': {'pyhap'},
 'homekit_controller': {'commentjson', 'aiohomekit'},
 'homematic': {'pyhomematic'},
 'homematicip_cloud': {'homematicip'},
 'homewizard': {'aiohwenergy'},
 'homeworks': {'pyhomeworks'},
 'honeywell': {'somecomfort'},
 'horizon': {'horimote'},
 'html5': {'pywebpush', 'py_vapid', 'http_ece'},
 'htu21d': {'i2csense', 'smbus'},
 'huawei_lte': {'huawei_lte_api'},
 'hue': {'aiohue'},
 'huisbaasje': {'huisbaasje'},
 'hunterdouglas_powerview': {'aiopvapi'},
 'hvv_departures': {'pygti'},
 'hydrawise': {'hydrawiser'},
 'hyperion': {'hyperion'},
 'ialarm': {'pyialarm'},
 'iammeter': {'iammeter'},
 'iaqualink': {'iaqualink'},
 'icloud': {'keyrings.alt', 'pyicloud'},
 'idteck_prox': {'rfk101py'},
 'ifttt': {'pyfttt'},
 'iglo': {'iglo'},
 'ign_sismologia': {'georss_ign_sismologia_client'},
 'ihc': {'ihcsdk'},
 'imap': {'aioimaplib'},
 'incomfort': {'incomfortclient'},
 'influxdb': {'influxdb', 'influxdb_client'},
 'insteon': {'pyinsteon', 'pypubsub'},
 'intellifire': {'intellifire4py'},
 'intesishome': {'pyintesishome'},
 'iotawatt': {'iotawattpy'},
 'iperf3': {'iperf3'},
 'ipma': {'geopy', 'pyipma'},
 'ipp': {'deepmerge', 'pyipp'},
 'iqvia': {'pyiqvia'},
 'irish_rail_transport': {'pyirishrail'},
 'islamic_prayer_times': {'prayer_times_calculator'},
 'iss': {'pyiss'},
 'isy994': {'pyisy'},
 'izone': {'pizone'},
 'jellyfin': {'jellyfin_apiclient_python'},
 'jewish_calendar': {'hdate'},
 'joaoapps_join': {'pyjoin'},
 'juicenet': {'pyjuicenet'},
 'kaiterra': {'kaiterra_async_client'},
 'keba': {'keba_kecontact'},
 'keenetic_ndms2': {'ndms2_client'},
 'kef': {'tenacity', 'aiokef'},
 'keyboard': {'pykeyboard'},
 'keyboard_remote': {'aionotify', 'evdev'},
 'kira': {'pykira'},
 'kiwi': {'kiwiki'},
 'kmtronic': {'pykmtronic'},
 'knx': {'xknx'},
 'kodi': {'jsonrpc_async', 'jsonrpc_websocket', 'jsonrpc_base', 'pykodi'},
 'konnected': {'konnected'},
 'kostal_plenticore': {'kostal'},
 'kraken': {'pykrakenapi', 'krakenex'},
 'kulersky': {'pykulersky', 'bleak'},
 'kwb': {'pykwb'},
 'lacrosse': {'pylacrosse'},
 'lametric': {'lmnotify'},
 'lastfm': {'pylast'},
 'launch_library': {'pylaunches'},
 'lcn': {'pypck'},
 'lg_netcast': {'pylgnetcast'},
 'lg_soundbar': {'temescal'},
 'life360': {'life360'},
 'lifx': {'aiolifx_effects', 'bitstring', 'aiolifx'},
 'lightwave': {'lightwave'},
 'limitlessled': {'limitlessled'},
 'linode': {'linode'},
 'linux_battery': {'batinfo'},
 'lirc': {'lirc'},
 'litejet': {'pylitejet'},
 'litterrobot': {'pylitterbot'},
 'logi_circle': {'logi_circle'},
 'london_underground': {'london_tube_status'},
 'lookin': {'aiolookin'},
 'luci': {'openwrt_luci_rpc'},
 'luftdaten': {'luftdaten'},
 'lupusec': {'lupupy'},
 'lutron': {'pylutron'},
 'lutron_caseta': {'pylutron_caseta'},
 'lyric': {'aiolyric'},
 'magicseaweed': {'magicseaweed'},
 'mailgun': {'pymailgunner'},
 'marytts': {'speak2mary'},
 'mastodon': {'mastodon'},
 'matrix': {'matrix_client'},
 'maxcube': {'maxcube'},
 'mazda': {'pymazda'},
 'mcp23017': {'adafruit_mcp230xx'},
 'media_extractor': {'youtube_dl'},
 'mediaroom': {'pymediaroom'},
 'melcloud': {'pymelcloud'},
 'melissa': {'melissa'},
 'message_bird': {'messagebird'},
 'met': {'metno'},
 'met_eireann': {'meteireann'},
 'meteo_france': {'meteofrance_api'},
 'meteoalarm': {'meteoalertapi'},
 'meteoclimatic': {'meteoclimatic'},
 'metoffice': {'datapoint'},
 'mfi': {'mficlient'},
 'mhz19': {'pmsensor'},
 'microsoft': {'pycsspeechtts'},
 'miflora': {'btlewrap', 'miflora'},
 'mikrotik': {'librouteros'},
 'mill': {'mill_local', 'mill'},
 'minecraft_server': {'mcstatus', 'dnspython'},
 'minio': {'minio'},
 'mitemp_bt': {'btlewrap', 'mitemp_bt'},
 'mobile_app': {'nacl', 'emoji'},
 'mochad': {'pbr', 'pymochad'},
 'modbus': {'pymodbus'},
 'modem_callerid': {'phone_modem'},
 'modern_forms': {'aiomodernforms'},
 'monoprice': {'pymonoprice'},
 'motion_blinds': {'motionblinds'},
 'motioneye': {'motioneye_client'},
 'mpd': {'mpd'},
 'msteams': {'pymsteams'},
 'mutesync': {'mutesync'},
 'mvglive': {'MVGLive'},
 'mycroft': {'mycroftapi'},
 'myq': {'pymyq', 'pkce'},
 'mysensors': {'mysensors'},
 'mystrom': {'pymystrom'},
 'mythicbeastsdns': {'mbddns'},
 'nad': {'nad_receiver'},
 'nam': {'nettigo_air_monitor'},
 'nanoleaf': {'aionanoleaf'},
 'neato': {'pybotvac'},
 'ness_alarm': {'nessclient'},
 'nest': {'nest', 'google_nest_sdm'},
 'netatmo': {'pyatmo'},
 'netdata': {'netdata'},
 'netgear': {'pynetgear'},
 'netgear_lte': {'eternalegypt'},
 'neurio_energy': {'neurio'},
 'nexia': {'nexia'},
 'nextbus': {'py_nextbus'},
 'nfandroidtv': {'notifications_android_tv'},
 'nightscout': {'py_nightscout'},
 'niko_home_control': {'nikohomecontrol'},
 'nilu': {'niluclient'},
 'nina': {'pynina'},
 'nissan_leaf': {'pycarwings2'},
 'nmap_tracker': {'nmap'},
 'nmbs': {'pyrail'},
 'noaa_tides': {'noaa_coops'},
 'norway_air': {'metno'},
 'notify_events': {'notify_events'},
 'notion': {'aionotion'},
 'nsw_fuel_station': {'nsw_fuel'},
 'nsw_rural_fire_service_feed': {'aio_geojson_nsw_rfs_incidents'},
 'nuheat': {'nuheat'},
 'nuki': {'pynuki'},
 'numato': {'numato_gpio'},
 'nut': {'pynut2'},
 'nws': {'pynws', 'metar'},
 'nx584': {'nx584'},
 'nzbget': {'pynzbgetapi'},
 'oasa_telematics': {'oasatelematics'},
 'obihai': {'pyobihai'},
 'octoprint': {'pyoctoprintapi'},
 'oem': {'oemthermostat'},
 'omnilogic': {'omnilogic', 'config'},
 'oncue': {'aiooncue'},
 'ondilo_ico': {'ondilo'},
 'onewire': {'pi1wire', 'pyownet'},
 'onkyo': {'eiscp'},
 'onvif': {'onvif', 'wsdiscovery', 'zeep'},
 'open_meteo': {'open_meteo'},
 'openerz': {'openerz_api'},
 'openevse': {'openevsewifi'},
 'opengarage': {'opengarage'},
 'openhome': {'async_upnp_client', 'openhomedevice'},
 'opensensemap': {'opensensemap_api'},
 'opentherm_gw': {'pyotgw'},
 'openuv': {'pyopenuv'},
 'openweathermap': {'pyowm', 'geojson', 'pysocks'},
 'opnsense': {'pyopnsense', 'pbr'},
 'opple': {'pyoppleio'},
 'orangepi_gpio': {'orangepi', 'OPi', 'nanopi'},
 'oru': {'oru'},
 'orvibo': {'orvibo'},
 'osramlightify': {'lightify'},
 'otp': {'pyotp'},
 'overkiz': {'pyhumps', 'pyoverkiz', 'boto3', 's3transfer', 'botocore'},
 'ovo_energy': {'ovoenergy'},
 'owntracks': {'nacl'},
 'ozw': {'openzwavemqtt'},
 'p1_monitor': {'p1monitor'},
 'panasonic_bluray': {'panacotta'},
 'panasonic_viera': {'panasonic_viera'},
 'pandora': {'pexpect', 'ptyprocess'},
 'pcal9535a': {'pcal9535a', 'smbus_cffi'},
 'pencom': {'pencompy'},
 'philips_js': {'haphilipsjs'},
 'pi4ioe5v9xxxx': {'pi4ioe5v9xxxx', 'smbus2'},
 'pi_hole': {'hole'},
 'picnic': {'python_picnic_api'},
 'pilight': {'pilight'},
 'ping': {'icmplib'},
 'pjlink': {'pypjlink'},
 'plaato': {'pyplaato'},
 'plex': {'plexwebsocket', 'plexapi'},
 'plugwise': {'plugwise', 'crcmod'},
 'plum_lightpad': {'plumlightpad'},
 'pocketcasts': {'pycketcasts'},
 'point': {'pypoint'},
 'poolsense': {'poolsense'},
 'powerwall': {'tesla_powerwall'},
 'progettihwsw': {'ProgettiHWSW'},
 'proliphix': {'proliphix'},
 'prometheus': {'prometheus_client'},
 'prosegur': {'pyprosegur'},
 'proxmoxve': {'proxmoxer'},
 'ps4': {'pyps4_2ndscreen'},
 'pushbullet': {'pushbullet'},
 'pushover': {'pushover_complete'},
 'pvoutput': {'pvo'},
 'pvpc_hourly_pricing': {'holidays', 'aiopvpc'},
 'python_script': {'RestrictedPython'},
 'qbittorrent': {'qbittorrent'},
 'qld_bushfire': {'georss_qld_bushfire_alert_client'},
 'qnap': {'qnapstats'},
 'qrcode': {'pyzbar'},
 'qvr_pro': {'pyqvrpro'},
 'qwikswitch': {'pyqwikswitch'},
 'rachio': {'rachiopy'},
 'radiotherm': {'radiotherm'},
 'rainbird': {'pyrainbird'},
 'raincloud': {'raincloudy'},
 'rainforest_eagle': {'uEagle', 'aioeagle'},
 'rainmachine': {'regenmaschine'},
 'raspyrfm': {'raspyrfm_client'},
 'rdw': {'vehicle'},
 'recollect_waste': {'aiorecollect'},
 'recswitch': {'pyrecswitch'},
 'reddit': {'praw', 'prawcore'},
 'rejseplanen': {'rjpl'},
 'remember_the_milk': {'rtmapi'},
 'remote_rpi_gpio': {'gpiozero'},
 'renault': {'renault_api'},
 'repetier': {'pyrepetierng'},
 'rflink': {'rflink'},
 'rfxtrx': {'RFXtrx'},
 'ridwell': {'aioridwell'},
 'ring': {'ring_doorbell'},
 'ripple': {'pyripple'},
 'risco': {'pyrisco'},
 'rituals_perfume_genie': {'pyrituals'},
 'rmvtransport': {'RMVtransport'},
 'rocketchat': {'rocketchat_API'},
 'roku': {'rokuecp'},
 'roomba': {'roombapy', 'paho_mqtt'},
 'roon': {'roonapi'},
 'route53': {'boto3', 's3transfer', 'botocore'},
 'rova': {'rova'},
 'rpi_gpio': {'RPi'},
 'rpi_gpio_pwm': {'adafruit_blinka',
                  'adafruit_circuitpython_pca9685',
                  'pwmled'},
 'rpi_pfio': {'pifacedigitalio'},
 'rpi_power': {'rpi_bad_power'},
 'rtsp_to_webrtc': {'rtsp_to_webrtc'},
 'ruckus_unleashed': {'pexpect', 'pyruckus'},
 'russound_rio': {'russound_rio'},
 'russound_rnet': {'russound'},
 'sabnzbd': {'pysabnzbd'},
 'saj': {'pysaj'},
 'samsungtv': {'samsungtvws', 'samsungctl'},
 'satel_integra': {'satel_integra'},
 'schluter': {'schluter'},
 'screenlogic': {'screenlogicpy'},
 'scsgate': {'scsgate'},
 'season': {'ephem'},
 'sendgrid': {'sendgrid'},
 'sense': {'sense_energy'},
 'sensehat': {'sense_hat'},
 'senseme': {'aiosenseme'},
 'sensibo': {'pysensibo'},
 'serial_pm': {'pmsensor'},
 'sesame': {'pysesame2'},
 'seventeentrack': {'py17track'},
 'sharkiq': {'sharkiqpy'},
 'shelly': {'aioshelly'},
 'shiftr': {'paho'},
 'shodan': {'shodan'},
 'sia': {'pysiaalarm'},
 'sighthound': {'simplehound'},
 'signal_messenger': {'pysignalclirestapi'},
 'simplepush': {'simplepush'},
 'simplisafe': {'simplipy'},
 'sinch': {'clx'},
 'sisyphus': {'sisyphus_control'},
 'sky_hub': {'pyskyqhub'},
 'skybeacon': {'pygatt'},
 'skybell': {'skybellpy'},
 'slack': {'slack'},
 'sleepiq': {'sleepyq'},
 'slide': {'goslideapi'},
 'sma': {'pysma'},
 'smappee': {'paho_mqtt', 'pysmappee'},
 'smart_meter_texas': {'smart_meter_texas'},
 'smarthab': {'pysmarthab'},
 'smartthings': {'pysmartthings', 'pysmartapp', 'httpsig'},
 'smarttub': {'smarttub'},
 'smarty': {'pysmarty', 'pymodbus'},
 'smhi': {'smhi'},
 'sms': {'gammu'},
 'snapcast': {'snapcast', 'construct'},
 'snmp': {'pysmi', 'pysnmp', 'pyasn1'},
 'sochain': {'pysochain'},
 'solaredge': {'solaredge'},
 'solaredge_local': {'solaredge_local'},
 'solarlog': {'sunwatcher'},
 'solax': {'solax'},
 'soma': {'api'},
 'somfy': {'pymfy'},
 'somfy_mylink': {'somfy_mylink_synergy'},
 'sonarr': {'sonarr'},
 'songpal': {'songpal'},
 'sonos': {'soco'},
 'sony_projector': {'pysdcp'},
 'soundtouch': {'libsoundtouch'},
 'spc': {'pyspcwebgw'},
 'spider': {'spiderpy'},
 'splunk': {'hass_splunk'},
 'spotify': {'spotipy'},
 'squeezebox': {'pysqueezebox'},
 'srp_energy': {'srpenergy'},
 'ssdp': {'async_upnp_client'},
 'starline': {'starline'},
 'starlingbank': {'starlingbank'},
 'statsd': {'statsd'},
 'steam_online': {'steam'},
 'steamist': {'discovery30303', 'aiosteamist'},
 'stiebel_eltron': {'pystiebeleltron', 'pymodbus'},
 'stookalert': {'stookalert'},
 'stream': {'av'},
 'streamlabswater': {'streamlabswater'},
 'subaru': {'subarulink', 'stdiomask'},
 'suez_water': {'regex', 'pysuez'},
 'supla': {'asyncpysupla'},
 'surepetcare': {'surepy', 'rich'},
 'swiss_hydrological_data': {'swisshydrodata'},
 'swiss_public_transport': {'opendata_transport'},
 'switchbot': {'switchbot'},
 'switcher_kis': {'aioswitcher'},
 'switchmate': {'switchmate'},
 'syncthing': {'aiosyncthing'},
 'syncthru': {'pysyncthru'},
 'synology_dsm': {'synology_dsm'},
 'synology_srm': {'synology_srm'},
 'system_bridge': {'systembridge'},
 'systemmonitor': {'psutil'},
 'tado': {'PyTado'},
 'tailscale': {'tailscale'},
 'tank_utility': {'tank_utility'},
 'tankerkoenig': {'pytankerkoenig'},
 'tapsaff': {'tapsaff'},
 'tasmota': {'hatasmota'},
 'tautulli': {'pytautulli'},
 'telegram_bot': {'telegram'},
 'tellstick': {'tellcore'},
 'temper': {'temperusb', 'pyusb'},
 'tensorflow': {'tensorflow'},
 'tesla_wall_connector': {'tesla_wall_connector'},
 'thermoworks_smoke': {'thermoworks_smoke'},
 'thingspeak': {'thingspeak'},
 'thinkingcleaner': {'pythinkingcleaner'},
 'tibber': {'tibber'},
 'tikteck': {'tikteck'},
 'tile': {'pytile'},
 'tmb': {'tmb'},
 'todoist': {'todoist'},
 'tof': {'VL53L1X2'},
 'tolo': {'tololib'},
 'toon': {'toonapi'},
 'totalconnect': {'total_connect_client'},
 'touchline': {'pytouchline'},
 'tplink': {'kasa'},
 'tplink_lte': {'tp_connected'},
 'traccar': {'pytraccar'},
 'tractive': {'aiotractive'},
 'tradfri': {'pytradfri'},
 'trafikverket_train': {'pytrafikverket'},
 'trafikverket_weatherstation': {'pytrafikverket'},
 'transmission': {'transmissionrpc'},
 'transport_nsw': {'TransportNSW'},
 'travisci': {'travispy'},
 'tts': {'mutagen'},
 'tuya': {'tuya_iot'},
 'twentemilieu': {'twentemilieu'},
 'twilio': {'twilio'},
 'twilio_call': {'twilio'},
 'twinkly': {'ttls'},
 'twitch': {'twitch'},
 'twitter': {'TwitterAPI'},
 'ubus': {'openwrt'},
 'unifi': {'aiounifi'},
 'unifi_direct': {'pexpect', 'ptyprocess'},
 'unifiled': {'unifiled'},
 'unifiprotect': {'pyunifiprotect', 'unifi_discovery'},
 'upb': {'upb_lib'},
 'upc_connect': {'connect_box'},
 'upcloud': {'upcloud_api'},
 'upnp': {'async_upnp_client'},
 'uptimerobot': {'pyuptimerobot'},
 'uscis': {'uscisstatus'},
 'usgs_earthquakes_feed': {'geojson_client'},
 'utility_meter': {'croniter'},
 'uvc': {'uvcclient'},
 'vallox': {'vallox_websocket_api'},
 'vasttrafik': {'vasttrafik'},
 'velbus': {'velbusaio'},
 'velux': {'pyvlx'},
 'venstar': {'venstarcolortouch'},
 'vera': {'pyvera'},
 'verisure': {'verisure'},
 'versasense': {'pyversasense'},
 'version': {'pyhaversion'},
 'vesync': {'pyvesync'},
 'vicare': {'PyViCare'},
 'vilfo': {'vilfo'},
 'vivotek': {'libpyvivotek'},
 'vizio': {'pyvizio'},
 'vlc_telnet': {'aiovlc'},
 'volkszaehler': {'volkszaehler'},
 'volumio': {'pyvolumio'},
 'volvooncall': {'volvooncall', 'hbmqtt', 'geopy'},
 'vultr': {'vultr'},
 'w800rf32': {'W800rf32'},
 'wallbox': {'wallbox'},
 'waqi': {'waqiasync'},
 'waterfurnace': {'waterfurnace'},
 'watson_iot': {'paho_mqtt', 'ibmiotf'},
 'watson_tts': {'ibm_watson', 'ibm_cloud_sdk_core'},
 'watttime': {'aiowatttime'},
 'waze_travel_time': {'WazeRouteCalculator'},
 'webostv': {'aiowebostv'},
 'wemo': {'pywemo'},
 'whirlpool': {'whirlpool'},
 'whois': {'whois'},
 'wiffi': {'wiffi'},
 'wilight': {'pywilight'},
 'wirelesstag': {'wirelesstagpy'},
 'withings': {'withings_api'},
 'wled': {'wled'},
 'wolflink': {'wolf_smartset'},
 'workday': {'convertdate',
             'hijri_converter',
             'holidays',
             'korean_lunar_calendar'},
 'xbee': {'xbee_helper'},
 'xbox_live': {'xboxapi'},
 'xeoma': {'pyxeoma'},
 'xiaomi_aqara': {'xiaomi_gateway'},
 'xiaomi_miio': {'miio', 'construct', 'micloud'},
 'xiaomi_tv': {'pymitv'},
 'xmpp': {'pyasn1', 'slixmpp'},
 'xs1': {'xs1_api_client'},
 'yale_smart_alarm': {'yalesmartalarmclient'},
 'yamaha': {'rxv'},
 'yamaha_musiccast': {'aiomusiccast'},
 'yeelight': {'async_upnp_client', 'yeelight'},
 'yeelightsunflower': {'yeelightsunflower'},
 'yi': {'aioftp'},
 'youless': {'youless_api'},
 'zabbix': {'pyzabbix'},
 'zengge': {'zengge'},
 'zerproc': {'pyzerproc', 'bleak'},
 'zha': {'aiosqlite',
         'bellows',
         'crccheck',
         'pure_pcapy3',
         'zhaquirks',
         'zigpy',
         'zigpy_deconz',
         'zigpy_xbee',
         'zigpy_zigate',
         'zigpy_znp'},
 'zhong_hong': {'zhong_hong_hvac'},
 'zoneminder': {'zoneminder'},
 'zwave_js': {'zwave_js_server'}}
```

thecode commented:
Shelly, Switcher and webostv looks fine

bdraco commented:
I used `homeassistant.util.json` to rewrite these files.  I noticed all the formatting wasn't consistent so if someone has preferred json flags, I can run it again.

jjlawren reviewed (approved):
Sonos and Plex are good 👍

mib1185 reviewed (approved):
Fine for Fritz Tools, Fritz Smarthome and Synology DSM

balloob commented:
how did you find the loggers?

bdraco commented:
> how did you find the loggers?

```
def get_imports(path) -> Generator[Import]:
    with open(path) as fh:
        root = ast.parse(fh.read(), path)

    for node in ast.iter_child_nodes(root):
        if isinstance(node, ast.Import):
            module = []
            level = 0
        elif isinstance(node, ast.ImportFrom) and node.module:
            module = node.module.split(".")
            level = node.level
        else:
            continue

        for n in node.names:
            yield Import(module, n.name.split("."), n.asname, level)


def get_level_zero_imports(integration: str) -> set[str]:
    python_files = component_dir.glob(f"{integration}/**/*.py")
    modules = set()
    for py_file in python_files:
        for py_import in get_imports(py_file):
            if py_import.level != 0:
                continue
            if py_import.module:
                module = py_import.module[0]
            elif py_import.name:
                module = py_import.name[0]
            if module in EXCLUDE_MODULES:
                continue
            modules.add(module)
    return modules
```

Kane610 commented:
Shouldn't debug for the integration be enabled as well?

bdraco commented:
> Shouldn't debug for the integration be enabled as well?

Yes, but we can impute that on the fly so it didn't seem like we need to store it

frenck commented:
Doesn't this assume the package name is used for logging? That doesn't have to be the case...

bdraco commented:
> Doesn't this assume the (external) package name is used for logging? That doesn't have to be the case...

Some of them may need to be adjusted but it shouldn't cause any problems if they are inconstantly named.  

The behavior is the same as if they were not there and nothing is preventing the codeowner from adjusting them later.

frenck commented:
Right, but that does mean we will be accepting something for which, will already know, it will most likely contain errors...

While cool, I rather see it handled by the codeowners/maintainers in that case.

Swamp-Ig commented:
Looks good for izone.

bdraco committed:
Drop manifest changes

bdraco commented:
> My opinion on the matter is: If we don't know it's correct, we should not add it.

I went ahead and dropped the manifest changes.  The list is above so code owners can use that if its correct and open individual PRs

frenck commented:
> I went ahead and dropped the manifest changes

Please note, I was sharing my opinion on it (and explicitly stating it was my opinion for that reason). I might not be in the majority in this.

rytilahti commented:
`xiaomi_miio` should not include `construct` as that is a very low-level lib, looks good for `songpal` and `tplink`.

My personal opinion contradicts with Franck's notion: accidentally enabling incorrect/unavailable loggers is much better situation to be in than not having any debug output, or am I not seeing what downsides this could have? Considering the large number of integrations this affects, it would take a loooooong time to get all codeowners to update their integrations to do this.

marvin-w reviewed (approved):
Current changes look good and work fine locally! Thanks @bdraco.

I'd have changed the KNX loggers previously as I'd have preferred them to be a little bit more granular.

https://www.home-assistant.io/integrations/knx/#logs-for-the-knx-integration

```
        "xknx.log",
        "xknx.raw_socket",
        "xknx.knx",
        "xknx.telegram",
        "xknx.state_updater"
```

Of course that depends on what should and will happen with this information.

 I **personally** have no hard feelings on merging this initially.

elupus commented:
Looks good from Philips J's and arcam_fmj. I'm also of the opinion that this does not make anything worse if the logger is some bit off, but it makes things much much better for the majority of systems. If some of them are somewhat off, code owner can fix that later.

raman325 commented:
> Looks good from Philips J's and arcam_fmj. I'm also of the opinion that this does not make anything worse if the logger is some bit off, but it makes things much much better for the majority of systems. If some of them are somewhat off, code owner can fix that later.

I vote to re-add the manifest changes for the same reason. We should add a blog post to the developers site about this for awareness in either case

balloob commented:
I think that it's fine to add to the manifests even if it's only 99% correct.

I have high confidence that it's correct because Python convention is to name loggers after your package module (`logging.getLogger(__name__)` so looking at imports should be correct.

I also think that trying to include too many packages wouldn't hurt, a person looking at the logs could always decide to hide it if it's not relevant. But even in the case of `construct` above, it is still used and could still be relevant.

bdraco committed:
preserve formatting

bdraco commented:
I pushed it again with some tweaks to ABC sort the list, and preserve formatting of the original json.

azogue reviewed (approved):
Looks good for `pvpc_hourly_pricing` 👍

uvjustin commented:
For `stream` the external loggers are actually `libav.*`, but those levels come in all wrong anyway. In `stream` we actually look to see if the internal logger is set to debug and then adjust those `libav.*` levels accordingly. So for this PR, `av` can be removed from the manifest for `stream`.
However, since the internal logger level can now change, we'll have to make an adjustment to modify the `libav.*` logger levels every time the internal logger level is changed. It's probably easiest for us to do this in a separate PR.

Santobert reviewed (approved):
LGTM for neato :+1:

mib1185 reviewed (approved):
Fine for `fritz`, `fritzbox`and `synology_dsm` 👍

allenporter reviewed (approved):
Seems fine for `nest`, `rtsp_to_webrtc`.

frenck commented on homeassistant/components/rdw/manifest.json:
```diff
@@ -6,5 +6,6 @@
   "quality_scale": "platinum",
-  "iot_class": "cloud_polling"
+  "iot_class": "cloud_polling",
+  "loggers": ["vehicle"]
```
This logger does not exists

frenck commented on homeassistant/components/tailscale/manifest.json:
```diff
@@ -6,5 +6,6 @@
   "quality_scale": "platinum",
-  "iot_class": "cloud_polling"
+  "iot_class": "cloud_polling",
+  "loggers": ["tailscale"]
```
This logger does not exists

frenck commented on homeassistant/components/ambee/manifest.json:
```diff
@@ -6,5 +6,6 @@
   "codeowners": ["@frenck"],
   "quality_scale": "platinum",
-  "iot_class": "cloud_polling"
+  "iot_class": "cloud_polling",
+  "loggers": ["ambee"]
```
This logger does not exist

frenck commented on homeassistant/components/cpuspeed/manifest.json:
```diff
@@ -5,5 +5,6 @@
   "codeowners": ["@fabaff", "@frenck"],
-  "iot_class": "local_push"
+  "iot_class": "local_push",
+  "loggers": ["cpuinfo"]
```
This logger does not exists

frenck commented on homeassistant/components/elgato/manifest.json:
```diff
@@ -7,5 +7,6 @@
   "quality_scale": "platinum",
-  "iot_class": "local_polling"
+  "iot_class": "local_polling",
+  "loggers": ["elgato"]
```
This logger does not exist

frenck commented on homeassistant/components/forecast_solar/manifest.json:
```diff
@@ -6,5 +6,6 @@
   "quality_scale": "platinum",
-  "iot_class": "cloud_polling"
+  "iot_class": "cloud_polling",
+  "loggers": ["forecast_solar"]
```
This logger does not exist

frenck commented on homeassistant/components/frontier_silicon/manifest.json:
```diff
@@ -4,5 +4,6 @@
   "codeowners": [],
-  "iot_class": "local_push"
+  "iot_class": "local_push",
+  "loggers": ["afsapi"]
```
This logger does not exist

frenck commented on homeassistant/components/futurenow/manifest.json:
```diff
@@ -4,5 +4,6 @@
   "codeowners": [],
-  "iot_class": "local_polling"
+  "iot_class": "local_polling",
+  "loggers": ["pyfnip"]
```
This logger does not exist

frenck commented on homeassistant/components/garages_amsterdam/manifest.json:
```diff
@@ -5,5 +5,6 @@
   "codeowners": ["@klaasnicolaas"],
-  "iot_class": "cloud_polling"
+  "iot_class": "cloud_polling",
+  "loggers": ["garages_amsterdam"]
```
This logger does not exist

frenck commented on homeassistant/components/gdacs/manifest.json:
```diff
@@ -6,5 +6,6 @@
   "quality_scale": "platinum",
-  "iot_class": "cloud_polling"
+  "iot_class": "cloud_polling",
+  "loggers": ["aio_georss_gdacs"]
```
This logger does not exist

frenck commented on homeassistant/components/launch_library/manifest.json:
```diff
@@ -5,5 +5,6 @@
   "codeowners": ["@ludeeus", "@DurgNomis-drol"],
-  "iot_class": "cloud_polling"
+  "iot_class": "cloud_polling",
+  "loggers": ["pylaunches"]
```
This logger does not exist

frenck commented on homeassistant/components/open_meteo/manifest.json:
```diff
@@ -6,5 +6,6 @@
   "codeowners": ["@frenck"],
-  "iot_class": "cloud_polling"
+  "iot_class": "cloud_polling",
+  "loggers": ["open_meteo"]
```
This logger does not exist

frenck commented on homeassistant/components/pvoutput/manifest.json:
```diff
@@ -6,5 +6,6 @@
   "iot_class": "cloud_polling",
-  "quality_scale": "platinum"
+  "quality_scale": "platinum",
+  "loggers": ["pvo"]
```
This logger does not exist

frenck commented on homeassistant/components/stookalert/manifest.json:
```diff
@@ -5,5 +5,6 @@
   "requirements": ["stookalert==0.1.4"],
-  "iot_class": "cloud_polling"
+  "iot_class": "cloud_polling",
+  "loggers": ["stookalert"]
```
This logger does not exist

frenck commented on homeassistant/components/aftership/manifest.json:
```diff
@@ -6,5 +6,6 @@
   "codeowners": [],
-  "iot_class": "cloud_polling"
+  "iot_class": "cloud_polling",
+  "loggers": ["pyaftership"]
```
This logger does not exist

frenck commented on homeassistant/components/evil_genius_labs/manifest.json:
```diff
@@ -5,5 +5,6 @@
   "codeowners": ["@balloob"],
-  "iot_class": "local_polling"
+  "iot_class": "local_polling",
+  "loggers": ["pyevilgenius"]
```
This logger does not exist

frenck commented on homeassistant/components/wled/manifest.json:
```diff
@@ -7,5 +7,6 @@
   "quality_scale": "platinum",
-  "iot_class": "local_push"
+  "iot_class": "local_push",
+  "loggers": ["wled"]
```
This logger does not exist

bdraco committed:
remove non-existant loggers

bdraco committed:
drop construct xiaomi_miio

farmio reviewed (approved):
`fronius` is fine. 
For `knx` Marvin left a comment - we can address that when this is merged in an extra PR.

bdraco commented:
I think this is good to merge once we get a ✅  on the arch issue.  I'll start working on part 2
